ttnn/cpp/ttnn/tensor/tensor_ops.cpp

```diff
@@ -374,7 +374,9 @@ Tensor tensor_reshape(const Tensor& input_tensor, const ttnn::Shape& new_shape)
                     if (tensor.memory_config().memory_layout != TensorMemoryLayout::HEIGHT_SHARDED) {
                         DeviceStorage device_storage = std::get<T>(tensor.get_storage());
                         DeviceBuffer device_buffer = device_storage.get_buffer();
-                        device_buffer->set_page_size(new_shape[-1] * tensor.element_size());
+                        if (new_shape[-1] != 0) {
+                            device_buffer->set_page_size(new_shape[-1] * tensor.element_size());
+                        }
```

Comment:
Should we set the page size to some default value in this case? I wonder if by minimizing it we can reduce unnecessary memory allocation on device later. Also will help reduce strange edge cases

Reply:
Actually I don't know what page_size mean(could you give some doc link of page_size?), if it mean num of tiles then I think set 0 make sense, but set 0 will hang the machine. I'll double check why set 0 not work

Reply:
Oh it's not num of tiles, its tile's size if TILE_LAYOUT(https://github.com/tenstorrent/tt-metal/blob/main/tech_reports/tensor_layouts/tensor_layouts.md#3-tensor-layout)

If set 0 the case of `input_shape, output_shape = [0], [1, 0]` will hang
and I print its value is
```
tensor.element_size(): 2
device_buffer->page_size(): 4
```
So currently I set page size to `2 * tensor.element_size()`

I try to figure out why its page_size is 4, I trace the code to here
https://github.com/tenstorrent/tt-metal/blob/61420b06c3fbc0d4e50e13e7aff0f60fb2f7714e/ttnn/cpp/ttnn/tensor/tensor_ops.cpp#L42
inside the `populate_buffers_and_metadata` the print is
```
other.element_size(): 2
other device_buffer->page_size(): 4
```
but in `tensor_to` I cannot print `async_safe_tensor`'s `page_size`, so I don't know why its page size is 2

Reply:
This part is not that important, it seems a bit more involved so maybe just keep it as it is unless we find an issue down the road. The fact view doesn't work with your changes (below) is much more important